### PR TITLE
[RFC] vim-patch:8.0.0184

### PR DESCRIPTION
**Pull request by sohnryang**
Merged vim patch 8.0.0184

### CODE REVIEW

File: src/nvim/testdir/test_system.vim [hunk 1]
```diff
@@ -46,3 +46,40 @@ function! Test_System()
 
   call assert_fails('call system("wc -l", 99999)', 'E86:')
 endfunction
+
+function! Test_system_exmode()
+  let cmd=" -es -u NONE -c 'source Xscript' +q; echo $?"
```
Comment (justinmk) [inline src/nvim/testdir/test_system.vim]: try `--headless` instead of `-es` in all cases.

File: src/nvim/testdir/test_system.vim [hunk 2]
```diff
@@ -46,3 +46,40 @@ function! Test_System()
...
+  let a = system(v:progpath . cmd)
+  call assert_equal('0', a[0])
+  call assert_equal(0, v:shell_error)
+
+  " Error before try does set error flag.
+  call writefile(['call nosuchfunction()', 'try', 'call doesnotexist()', 'catch', 'endtry'], 'Xscript')
+  let a = system(v:progpath . cmd)
+  call assert_notequal('0', a[0])
+
+  let cmd=" -es -u NONE -c 'source Xscript' +q"
```
Comment (justinmk) [inline src/nvim/testdir/test_system.vim]: also here, and below ...

**Comment by sohnryang** [comment]
@justinmk When I use `--headless` insted of `-es`, the return value is `0` instead of `1`, which makes the test fail.

What's the difference between `--headless` and `-es`?

**Comment by sohnryang** [comment]
@justinmk ping

**Comment by justinmk** [comment]
Why is this marked RDY? The build didn't even pass...

**Comment by ckelsel** [comment]
```
From test_system.vim:
Found errors in Test_system_exmode():
function RunTheTest[13]..Test_system_exmode line 6: Expected '0' but got '\e'

Test results:


From test_system.vim:
Found errors in Test_system_exmode():
function RunTheTest[13]..Test_system_exmode line 6: Expected '0' but got '\e'
TEST FAILURE
make: *** [report] Error 1
make: Leaving directory `/home/travis/build/neovim/neovim/src/nvim/testdir'
```

**Comment by ckelsel** [comment]
you can also merge vim-8.0.0197.patch @sohnryang

**Comment by justinmk** [comment]
merged

**Pull request closed without merging**